1.0/articles/es6.md

```diff
@@ -0,0 +1,298 @@
+
+tags:
+- es2015
+- es6
+- javascript
+---
+
+<style>
+paper-button.blue {
+  background: #4285f4;
+  color: #fff;
+}
+paper-button.blue:hover {
+  background: #2a56c6;
+}
+</style>
+
+{% include authorship.html %}
+
+{% include toc.html %}
+
+## Introduction
+
+Web components evolve markup into something that's meaningful, maintainable, and highly modular. Thanks to these new API primitives, not only do we have improved ergonomics when building apps, but we gain better overall structure, design, and reusability.
+
+ES6 does the same thing, but for JavaScript. JavaScript is the core language of the web and it's about time it saw a makeover! ES6 brings powerful new primitives to the language, but they're mostly about developer convenience. The result of all the syntactical goodness? More legible, more maintainable, and more modular JavaScript.
+
+The analogy is simple: **Web Components : HTML :: ES6 : JS**. That is to say, what web components do for HTML, ES6 does for JavaScript 
+
+In this article we'll create a `<stock-ticker>` custom element using ES6 classes.
+First, I'll cover how to create the element using the vanilla web component JS APIs and
+then how to create the same element using Polymer.
+
+<p layout horizontal center-center>
+<a href="http://ebidel.github.io/polymer-experiments/polymersummit/stockticker/" target="_blank">
+  <paper-button raised class="blue">Stock Ticker Demo</paper-button>
+</a>
+<a href="https://github.com/ebidel/polymer-experiments/tree/master/polymersummit/stockticker" target="_blank">
+  <paper-button raised class="blue">Full source</paper-button>
+</a>
+</p>
+
+## Defining custom elements from a class
+
+Custom elements can be defined from an ES6 class by extending the `HTMLElement` DOM interface:
+
+    'use strict';
+
+    class StockTicker extends HTMLElement {
+      ...
+    }
+
+Extending `HTMLElement` creates the element with the correct `prototype`, inheriting all the methods/properties of the DOM interface.
+
+**Note** Some browsers require `'use strict';` for using ES6 features but I'll be
+leaving it off the other examples on this page.
+{: .alert .alert-info }
+
+### Element "constructors" 
+
+Normally when creating a class, you define a `constructor` to do initialization work. However,
+in the world of custom elements this doesn't apply. Instead, you'll need to use
+the `createdCallback` [custom element lifecycle method](../docs/devguide/registering-elements.html#lifecycle-callbacks) for setup work. Use it to initialize properties, (optionally) create [Shadow DOM](http://www.html5rocks.com/tutorials/webcomponents/shadowdom/) for the element, set default attributes, etc.
+
+The following example creates a class definition for our stocker ticker element:
+
+    class StockTicker extends HTMLElement {
+
+      // Use createdCallback instead of constructor to init an element.
+      createdCallback() {
+        // This element uses Shadow DOM.
+        this.createShadowRoot().innerHTML = `
```

Comment:
Is this the template string? You mention it below and it's super hard to spot in code if it's just a pair of backticks.

I'd suggest introducing the idea here, like // set HTML using a template string, explained below

At the risk of adding length, might even want to break this into two lines, so it's:

         // This element uses Shadow DOM.
         var shadowRoot = this.createShadowRoot();
          // Initialize shadow DOM using a template string (discussed below)
          shadowRoot.innerHTML = `